rules/no-array-reduce.js

```diff
@@ -34,14 +36,50 @@ const selector = matches([
 	].join(''),
 ]);
 
-const create = () => {
+const schema = [
+	{
+		type: 'object',
+		properties: {
+			allowSimpleOperations: {
+				type: 'boolean',
+				default: true,
+			},
+		},
+	},
+];
+
+const create = context => {
+	const {allowSimpleOperations} = {allowSimpleOperations: true, ...context.options[0]};
+
 	return {
 		[selector](node) {
-			return {
+			const callback = get(node, 'parent.parent.arguments[0]', {});
+			const problem = {
 				node,
 				messageId: MESSAGE_ID,
 				data: {method: node.name},
 			};
+
+			if (!allowSimpleOperations) {
+				return problem;
+			}
+
+			if (callback.type === 'ArrowFunctionExpression' && callback.body.type === 'BinaryExpression') {
+				return;
+			}
+
+			if ((callback.type === 'ArrowFunctionExpression' || callback.type === 'FunctionExpression') &&
+				callback.body.type === 'BlockStatement' &&
+				callback.body.body[0].type === 'ReturnStatement' &&
```

Comment:
```suggestion
				callback.body.body.length === 1 &&
				callback.body.body[0].type === 'ReturnStatement' &&
```

I think we should only allow one `return`

```js
foo.reduce(function (a, b) {
	return a + doingComplicatedThings();

	function doingComplicatedThings() {
       // ...
	}
});
```

Reply:
Make sure to add a test for this too.

Reply:
Excellent idea / catch! Length check added along with corresponding test.